examples/jsm/nodes/lighting/LightingContextNode.js

```diff
@@ -55,21 +54,24 @@ class LightingContextNode extends ContextNode {
 		};
 
 		const lighting = {
-			radiance: temp( vec3() ),
-			irradiance: temp( vec3() ),
-			iblIrradiance: temp( vec3() ),
-			ambientOcclusion: temp( float( 1 ) )
+			radiance: vec3().temp(),
+			irradiance: vec3().temp(),
+			iblIrradiance: vec3().temp(),
+			ambientOcclusion: float( 1 ).temp()
 		};
 
+		context.reflectedLight = reflectedLight;
+		context.lightingModelNode = lightingModelNode || context.lightingModelNode;
+
 		Object.assign( properties, reflectedLight, lighting );
 		Object.assign( context, lighting );
 
-		context.reflectedLight = reflectedLight;
-		context.lightingModelNode = lightingModelNode || context.lightingModelNode;
+		// @TODO: Call needed return a new node ( or rename the ShaderNodeInternal.call() function ), it's not moment to run
```

Comment:
Not sure if I understand this comment... but can't we do `shader( ( {}, stack, builder ) => lightingModel.init.call( context, stack, builder ) )`, similar as how it was done in other places?

Reply:
The syntax `shader( ( {}, stack, builder )` is correct, but the `call` syntax should be just `.call( { ...inputs } )` like `FunctionCallNode` for the users and not `.call( {}, stack, builder )`. It because `call()` should return a `ShaderNodeCall` and not run the function immediately. When it's called in `construct()` process, we can send `stack` and `builder` parameters, but if the user calls the shader() outside `construct()` like `material.colorNode = shader(..).call()` we would not be able to pass the `builder/stack` because it is in the wrong builder stage.